components/mongodb/mongodb.app.mjs

```diff
@@ -1,11 +1,286 @@
   app: "mongodb",
-  propDefinitions: {},
+  propDefinitions: {
+    collection: {
+      label: "Collection",
+      type: "string",
+      description: "Collection where the new document will be inserted.",
+      async options() {
+        return this.getCollections();
+      },
+    },
+    document: {
+      label: "Document",
+      type: "string",
+      description: "The document ID",
+      async options({
+        collection,
+        page,
+      }) {
+        const LIMIT = 100;
+        const documents =  await this.searchDocuments(
+          collection,
+          undefined,
+          page * LIMIT,
+          LIMIT,
+        );
+        return documents.filter((doc) => doc._id).map((doc) => ({
+          label: doc.name || doc.title ?
+            `${doc.name || doc.title} - ObjectId(${doc._id})`
+            : `ObjectId(${doc._id})`,
+          value: doc._id,
+        }));
+      },
+    },
+    data: {
+      label: "Data",
+      type: "object",
+      description: "The object to be used in document creation. Dates must follow `yyyy-mm-dd` format",
+    },
+    parseNumbers: {
+      label: "Parse Numbers",
+      type: "boolean",
+      description: "If `true`, each number value represented by a string will be parsed to its respective type",
+      default: true,
+    },
+    parseBooleans: {
+      label: "Parse Booleans",
+      type: "boolean",
+      description: "If `true`, each boolean value represented by a string will be parsed to its respective type",
+      default: true,
+    },
+    parseDates: {
+      label: "Parse Dates",
+      type: "boolean",
+      description: "If `true`, each date value represented by a string will be parsed to its respective type",
+      default: true,
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async connect() {
+      const {
+        username,
+        password,
+        database,
+        hostname,
+      } = this.$auth;
+      const uri = `mongodb+srv://${username}:${password}@${hostname}/${database}?retryWrites=true&w=majority`;
+      await mongoose.connect(uri);
+    },
+    getCollections() {
+      return new Promise((resolve, reject) => {
+        this.connect()
+          .then(async () => {
+            try {
+              const collections = await mongoose.connection.db.listCollections().toArray();
+              resolve(collections.map((collection) => collection.name).sort());
+            } catch (err) {
+              reject(err);
+            } finally {
+              mongoose.connection.close();
+            }
+          })
+          .catch((err) => reject(err));
+      });
+    },
+    createDocument(data, collection, parseNumbers, parseBooleans, parseDates) {
+      return new Promise((resolve, reject) => {
+        this.connect().then(async () => {
+          try {
+            const Model = mongoose.model(collection, this.getSchemaByData(data));
```

Comment:
I believe this is the source of https://pipedream-users.slack.com/archives/CPTJYRY5A/p1651266007759259 . See https://nesin.io/blog/fix-mongoose-cannot-overwrite-model-once-compiled-error . You may need to memoize the model and return the model directly instead of calling `mongoose.model` again.

Reply:
@vellames-turing are you able to reproduce?